numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4088,6 +4088,7 @@ initialize_numeric_types(void)
      * #NAME = Byte, Short, Int, Long, LongLong, UByte, UShort,
      *         UInt, ULong, ULongLong#
      */
+    @name@_arrtype_as_number = gentype_as_number;
```

Comment:
On 15/07/16 09:16, Nathaniel J. Smith wrote:
>
> In numpy/core/src/multiarray/scalartypes.c.src 
> <https://github.com/numpy/numpy/pull/7839#discussion_r70979857>:
>
> > @@ -4088,6 +4088,7 @@ initialize_numeric_types(void)
> >       * #NAME = Byte, Short, Int, Long, LongLong, UByte, UShort,
> >       *         UInt, ULong, ULongLong#
> >       */
> > +    @name@_arrtype_as_number = gentype_as_number;
>
> On my phone so haven't looked at the context here, but do I infer 
> correctly from the commit message that this is just filling in some 
> arbitrary pointer here that will be overwritten later? If so then 
> there should be a comment to that effect here, because otherwise to a 
> naive reader it looks like there's some actual reason why 
> gentype_as_number will work.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/numpy/numpy/pull/7839/files/a4087dad3b1f1c7dcfb9db33dd8ff53612ec468a#r70979857>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAySZ4-nEEUPH1vCV3_KCtn2fQ4ZDGnwks5qV5Y_gaJpZM4JNa6j>.
>
It is not arbitrary, the pointer is from the base class, i.e. until the 
NULL functions are overloaded, the gentype functions  be called.
So indeed, they will work and are exactly the functions called until 
overloaded.
Note later in the same function, tp_richcompare is likewise set to 
gentype_richcompare.

Matti